### PR TITLE
Fixed py-utils unittests in main branch

### PR DESCRIPTION
**Pull request by veerendra-simha-garikipati**
Signed-off-by: Veerendra Garikipati <veerendra.garikipati@seagate.com>

# Problem Statement
- [Py-utils] - Fix py-utils test failures in main branch

# Design
-  [CORTX-30743](https://jts.seagate.com/browse/CORTX-30743)

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing
- Are test cases updated along with code changes due to Enhancements/Bugs [Y/N]: Y
- Confirm that new test cases are added to regression and sanity plan files and relevant feature plan files [Y/N]: Y
- Confirm that Test Cases are added for new features added [Y/N]: 
- Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- Confirm Testing was performed with installed RPM/K8s deployment [Y/N]: Y

# Review Checklist 
####  Before posting the PR please ensure:
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [ ] New package/s added to setup.py?
- [x] Jira is updated
- [ ] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified.

# KVstore/Confstore feature (changes/fixes) checklist
#### Confirm changes are made for:
- [x] ConfStore
- [ ] KVStore
- [ ] ConfCli
- [ ] Test cases added for all above 3
- [ ] changes done in all the KVpayloads (ConsulKvPayload, IniKvPayload, KvPayload)

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/test/plans/regression.pln [hunk 1]
```diff
@@ -5,32 +5,7 @@ cli_framework.test_cli_framework
...
-conf_store.test_conf_cli.TestConfCli.test_conf_cli_multiple_kv_delim
-conf_store.test_conf_cli.TestConfCli.test_conf_cli_kv_delim_no_value
-conf_store.test_conf_cli.TestConfCli.test_conf_cli_multiple_value_split_kv_delim
-conf_store.test_conf_cli.TestConfCli.test_conf_cli_with_kv_delim_in_value
-conf_store.test_conf_cli.TestConfCli.test_conf_cli_properties_wrong_format_kv
-conf_store.test_conf_cli.TestConfCli.test_conf_cli_by_get_diff_keys
-conf_store.test_conf_cli.TestConfCli.test_conf_cli_merge_keys
-conf_store.test_conf_cli.TestConfCli.test_conf_cli_merge
-
+conf_store.test_conf_cli
```
Comment (veerendra-simha-garikipati) [inline py-utils/test/plans/regression.pln:8]: Instead of running individual tests from test_conf_cli, we just run all the tests from test_script

Comment (rahul-tripathi-git) [reply py-utils/test/plans/regression.pln:8]: Why are we doing this now? 
Is this a limitation that got introduced now or is it an improvement?

**Review by rohit-k-dwivedi — APPROVED** [review]

File: py-utils/src/utils/conf_store/conf_store.py [hunk 2]
```diff
@@ -281,10 +281,7 @@ def get(index: str, key: str, default_val: str = None, **filters):
     @staticmethod
     def delete(index: str, key: str):
         """ Deletes a given key from the config """
-        is_deleted = Conf._conf.delete(index, key)
```
Comment (rohit-k-dwivedi) [inline py-utils/src/utils/conf_store/conf_store.py:284]: wondering if directly pushing this to main would affect other teams code who are still assuming delete automatically saves the file
cc @rahul-tripathi-git

Comment (cdeshmukh) [reply py-utils/src/utils/conf_store/conf_store.py:284]: agree with @rohit-k-dwivedi , why we are changing the behavior here @veerendra-simha-garikipati

Comment (rahul-tripathi-git) [reply py-utils/src/utils/conf_store/conf_store.py:284]: Hi @rohit-k-dwivedi 
1N-K8s deployment got passed, which indicated we are good. Right?

Comment (rohit-k-dwivedi) [reply py-utils/src/utils/conf_store/conf_store.py:284]: yes @rahul-tripathi-git we are good on the deployment front.

**Review by selvakumaar5496 — APPROVED** [review]

File: py-utils/src/utils/conf_store/conf_store.py (hunk 2 above)
Comment (rahul-tripathi-git) [reply py-utils/src/utils/conf_store/conf_store.py:284]: Hi @cdeshmukh 
This behavior was wrong and this change should not have been merged in the first place. Please consider this as a correction.
As per the defined behavior of the Conf class, any changes done in KV store by Conf.delete should not be saved back to the source file. We have a separate class, [MappedConf](https://github.com/Seagate/cortx-utils/blob/f6418409039a58dc73760324b2a5758df0988444/py-utils/src/utils/conf_store/conf_store.py#L348), for this.

Comment (cdeshmukh) [reply py-utils/src/utils/conf_store/conf_store.py:284]: makes sense

**Review by cdeshmukh — APPROVED** [review]

**Pull request merged**